src/GafferScene/EditScopeAlgo.cpp

```diff
@@ -854,3 +858,219 @@ const Gaffer::GraphComponent *GafferScene::EditScopeAlgo::attributeEditReadOnlyR
+		}
+
+		throw IECore::Exception( boost::str( boost::format( "Option \"%s\" does not exist" ) % option ) );
+	}
+
+	return result;
+}
+
+/// \todo Finding a child tweak plug by the tweak name is needed in a few different
+/// places (AttributeInspector, ParameterInspector). Consider adding this to TweaksPlug.
+
+TweakPlug *tweakPlug( TweaksPlug *tweaks, const std::string &tweakName )
+{
+	for( auto &p : TweakPlug::Range( *tweaks ) )
+	{
+		if( p->namePlug()->getValue() == tweakName )
+		{
+			return p.get();
+		}
+	}
+	return nullptr;
+}
+
+}  // namespace
+
+
+bool GafferScene::EditScopeAlgo::hasOptionEdit( const Gaffer::EditScope *scope, const std::string &option )
+{
+	return acquireOptionEdit( const_cast<EditScope *>( scope ), option, /* createIfNecessary = */ false );
+}
+
+TweakPlug *GafferScene::EditScopeAlgo::acquireOptionEdit( Gaffer::EditScope *scope, const std::string &option, bool createIfNecessary )
+{
+	// If we need to create an edit, we'll need to do a compute to figure our the option
+	// type and value. But we don't want to do that if we already have an edit. And since the
+	// compute could error, we need to get the option value before making _any_ changes, so we
+	// don't leave things in a partial state. We use `ensureOptionValue()` to get the value
+	// lazily at the first point we know it will be needed.
+	ConstDataPtr optionValue;
+	auto ensureOptionValue = [&] {
+		if( !optionValue )
+		{
+			optionValue = runTimeCast<const Data>(
+				::optionValue( scope->outPlug<ScenePlug>(), option )
+			);
+			if( !optionValue )
+			{
+				throw IECore::Exception( boost::str( boost::format( "Option \"%s\" cannot be tweaked" ) % option ) );
+			}
+		}
+	};
+
+	// Find processor
+
+	auto *processor = acquireOptionProcessor( scope, /* createIfNecessary */ false );
+	if( !processor )
+	{
+		if( !createIfNecessary )
+		{
+			return nullptr;
+		}
+		else
+		{
+			ensureOptionValue();
+			processor = acquireOptionProcessor( scope, /* createIfNecessary */ true );
+		}
+	}
+
+	auto *tweaks = processor->getChild<TweaksPlug>( "edits" );
+
+	if( TweakPlug *tweak = tweakPlug( tweaks, option ) )
+	{
+		return tweak;
+	}
+
+	if( !createIfNecessary )
+	{
+		return nullptr;
+	}
+
+	// No tweak for the option yet. Create it.
+
+	ensureOptionValue();
+
+	ValuePlugPtr valuePlug = PlugAlgo::createPlugFromData( "value", Plug::In, Plug::Default, optionValue.get() );
+
+	tweaks->addChild( new TweakPlug( option, valuePlug, TweakPlug::Create, false ) );
+
+	return tweakPlug( tweaks, option );
+}
+
+void GafferScene::EditScopeAlgo::removeOptionEdit( Gaffer::EditScope *scope, const std::string &option )
+{
+	TweakPlug *edit = acquireOptionEdit( scope, option, /* createIfNecessary */ false );
+	if( !edit )
+	{
+		return;
+	}
+
+	auto *processor = acquireOptionProcessor( scope, /* createIfNecessary */ false );
+	if( !processor )
+	{
+		return;
+	}
+
+	auto *tweaks = processor->getChild<TweaksPlug>( "edits" );
+	if( !tweaks )
+	{
+		return;
+	}
```

Comment:
Can't we just do `edit->parent()->removeChild( edit )`?

Reply:
Yes! Much nicer now, fixed in https://github.com/GafferHQ/gaffer/pull/4907/commits/978d11ccb8bb369d41e43b11e4715161c3becc2d